asserts/header_checks.go

```diff
@@ -132,13 +132,41 @@ func checkIntWhat(headers map[string]interface{}, name, what string) (int, error
 	if err != nil {
 		return -1, err
 	}
+	value, err := atoi(valueStr, "%q %s", name, what)
+	if err != nil {
+		return -1, err
+	}
+	return value, nil
+}
+
+type intSyntaxError string
+
+func (e intSyntaxError) Error() string {
+	return string(e)
+}
+
+func atoi(valueStr, whichFmt string, whichArgs ...interface{}) (int, error) {
 	value, err := strconv.Atoi(valueStr)
 	if err != nil {
-		return -1, fmt.Errorf("%q %s is not an integer: %v", name, what, valueStr)
+		which := fmt.Sprintf(whichFmt, whichArgs...)
+		if ne, ok := err.(*strconv.NumError); ok && ne.Err == strconv.ErrRange {
+			return -1, fmt.Errorf("%s is out of range: %v", which, valueStr)
+		}
+		return -1, intSyntaxError(fmt.Sprintf("%s is not an integer: %v", which, valueStr))
+	}
+	if !noPrefixZeros(valueStr, value == 0) {
+		return -1, fmt.Errorf("%s has invalid prefix zeros: %s", fmt.Sprintf(whichFmt, whichArgs...), valueStr)
 	}
 	return value, nil
 }
 
+func noPrefixZeros(s string, isZero bool) bool {
+	if strings.HasPrefix(s, "0") {
+		return isZero && s == "0"
+	}
```

Comment:
Could we simplify this like so and drop isZero?
```suggestion
	if strings.HasPrefix(s, "0") && s != "0" {
		return false
	}
```

Reply:
yes, possibly, what do you think of the rest of PR ? :)